src/core/public/i18n/i18n_service.tsx

```diff
@@ -23,16 +23,209 @@ import { EuiContext } from '@elastic/eui';
+        {
+          defaultMessage: "There aren't any options available",
+        }
+      ),
+      'euiComboBoxOptionsList.noMatchingOptions': ({ searchValue }: EuiValues) =>
+        i18n.translate('core.euiComboBoxOptionsList.noMatchingOptions', {
+          defaultMessage: "{searchValue} doesn't match any options",
+          values: { searchValue },
+        }),
+      'euiComboBoxPill.removeSelection': ({ children }: EuiValues) =>
+        i18n.translate('core.euiComboBoxPill.removeSelection', {
+          defaultMessage: 'Remove {children} from selection in this group',
+          values: { children },
+        }),
+      'euiForm.addressFormErrors': i18n.translate('core.euiForm.addressFormErrors', {
+        defaultMessage: 'Please address the errors in your form.',
+      }),
+      'euiFormControlLayoutClearButton.label': i18n.translate(
+        'core.euiFormControlLayoutClearButton.label',
+        {
+          defaultMessage: 'Clear input',
+        }
+      ),
+      'euiHeaderAlert.dismiss': i18n.translate('core.euiHeaderAlert.dismiss', {
+        defaultMessage: 'Dismiss',
+      }),
+      'euiHeaderLinks.appNavigation': i18n.translate('core.euiHeaderLinks.appNavigation', {
+        defaultMessage: 'App navigation',
+      }),
+      'euiHeaderLinks.openNavigationMenu': i18n.translate(
+        'core.euiHeaderLinks.openNavigationMenu',
+        {
+          defaultMessage: 'Open navigation menu',
+        }
+      ),
+      'euiModal.closeModal': i18n.translate('core.euiModal.closeModal', {
+        defaultMessage: 'Closes this modal window',
+      }),
+      'euiPagination.jumpToLastPage': ({ pageCount }: EuiValues) =>
+        i18n.translate('core.euiPagination.jumpToLastPage', {
+          defaultMessage: 'Jump to the last page, number {pageCount}',
+          values: { pageCount },
+        }),
+      'euiPagination.nextPage': i18n.translate('core.euiPagination.nextPage', {
+        defaultMessage: 'Next page',
+      }),
+      'euiPagination.pageOfTotal': ({ page, total }: EuiValues) =>
+        i18n.translate('core.euiPagination.pageOfTotal', {
+          defaultMessage: 'Page {page} of {total}',
+          values: { page, total },
+        }),
+      'euiPagination.previousPage': i18n.translate('core.euiPagination.previousPage', {
+        defaultMessage: 'Previous page',
+      }),
+      'euiPopover.screenReaderAnnouncement': i18n.translate(
+        'core.euiPopover.screenReaderAnnouncement',
+        {
+          defaultMessage: 'You are in a popup. To exit this popup, hit escape.',
+        }
+      ),
+      'euiStep.completeStep': i18n.translate('core.euiStep.completeStep', {
+        defaultMessage: 'Step',
+      }),
+      'euiStep.incompleteStep': i18n.translate('core.euiStep.incompleteStep', {
+        defaultMessage: 'Incomplete Step',
+      }),
+      'euiStepHorizontal.buttonTitle': ({ step, title, disabled, isComplete }: EuiValues) => {
+        let titleAppendix = '';
```

Comment:
__issue:__ `titleAppendix` should also be localized, I'd suggest using something like this (you can experiment **[here](https://format-message.github.io/icu-message-format-for-translators/editor.html?m=Step%20{step}:%20{title}{titleAppendix,%20select,%20completed%20{%20is%20completed}%20disabled%20{%20is%20disabled}%20other%20{}})**):
```ts
'euiStepHorizontal.buttonTitle': ({ step, title, disabled, isComplete }: EuiValues) => {
    return i18n.translate('core.euiStepHorizontal.buttonTitle', {
      defaultMessage:
        'Step {step}: {title}{titleAppendix, select, completed { is completed} disabled { is disabled} other {}}',
      values: {
        step,
        title,
        titleAppendix: disabled ? 'disabled' : isComplete ? 'completed' : '',
      },
    });
  },
```